BUILDING.md

```diff
@@ -0,0 +1,127 @@
+```
+
+## Setting Up Solvers
+
+ESBMC relies on SMT solvers to reason about formulae in its back-end.
+
+Currently we support the following solvers: __Boolector__, __CVC4__, __MathSAT__, __Yices__, __Yices (GMP)__, and __Z3__.
+
+Since this guide focuses primarily on ESBMC build, we will only cover the steps needed by it.
+
+### Setting Up Boolector
+
+We have wrapped the entire build and setup of Boolector in the following command:
+
+```
+git clone https://github.com/boolector/boolector && cd boolector && git reset --hard 3.2.0 && ./contrib/setup-lingeling.sh && ./contrib/setup-btor2tools.sh && ./configure.sh --prefix $PWD/../boolector-release && cd build && make -j9 && make install
+```
+
+If you need more details on Boolector, please refer to [its Github](https://github.com/Boolector/boolector).
+
+### Setting Up CVC4
+
+We have wrapped the entire build and setup of CVC4 in the following command:
+
+```
+wget https://github.com/CVC4/CVC4/archive/1.7.tar.gz && tar xf 1.7.tar.gz && rm 1.7.tar.gz && cd CVC4-1.7 && ./contrib/get-antlr-3.4 && ./configure.sh --optimized --prefix=../cvc4 --static --no-static-binary && cd build && make -j8 && make install
+```
+
+If you need more details on CVC4, please refer to [its Github](https://github.com/CVC4/CVC4).
+
+### Setting Up MathSAT
+
+We have wrapped the entire build and setup of MathSAT in the following command:
+
+```
+wget http://mathsat.fbk.eu/download.php?file=mathsat-5.5.4-linux-x86_64.tar.gz -O mathsat.tar.gz && tar xf mathsat.tar.gz && mv mathsat-5.5.4-linux-x86_64 mathsat
+```
+
+If you need more details on MathSAT, please refer to [its webpage](https://mathsat.fbk.eu).
+
+### Setting Up Yices
+
+First, we need to setup and build [GMP library](https://gmplib.org), by entering the following command:
+
+```
+wget https://gmplib.org/download/gmp/gmp-6.1.2.tar.xz && tar xf gmp-6.1.2.tar.xz && rm gmp-6.1.2.tar.xz && cd gmp-6.1.2 && ./configure --prefix $PWD/../gmp --disable-shared ABI=64 CFLAGS=-fPIC CPPFLAGS=-DPIC && make -j4 && make install
+```
+
+Then, we are able build and setup Yices using the following command:
+
+```
+git clone https://github.com/SRI-CSL/yices2.git && cd yices2 && git checkout Yices-2.6.1 && autoreconf -fi && ./configure --prefix $PWD/../yices --with-static-gmp=$PWD/../gmp/lib/libgmp.a && make -j9 && make static-lib && make install && cp ./build/x86_64-pc-linux-gnu-release/static_lib/libyices.a ../yices/lib
+```
+
+If you need more details on Yices, please refer to [its Github](https://github.com/SRI-CSL/yices2).
+
+### Setting Up Z3
+
+We have wrapped the entire build and setup of Z3 in the following command:
+
+```
+wget https://github.com/Z3Prover/z3/releases/download/z3-4.8.4/z3-4.8.4.d6df51951f4c-x64-ubuntu-16.04.zip && unzip z3-4.8.4.d6df51951f4c-x64-ubuntu-16.04.zip && mv z3-4.8.4.d6df51951f4c-x64-ubuntu-16.04 z3
+```
+
+If you need more details on Z3, please refer to [its Github](https://github.com/Z3Prover/z3).
+
+## Building ESBMC
```

Comment:
I think this section could explain more about the cmake options for esbmc. Or at least point to the [options](https://github.com/esbmc/esbmc/blob/master/scripts/cmake/Options.cmake) module